dynamiq/storages/vector/pgvector/pgvector.py

```diff
@@ -0,0 +1,551 @@
+from contextlib import contextmanager
+from enum import Enum
+from typing import Any, Optional
+
+import psycopg
+from pgvector.psycopg import register_vector
+from psycopg import Cursor
+from psycopg.rows import dict_row
+from psycopg.sql import SQL, Identifier
+from psycopg.sql import Literal as SQLLiteral
+from psycopg.types.json import Jsonb
+
+from dynamiq.connections import PGVector
+from dynamiq.storages.vector.exceptions import VectorStoreException
+from dynamiq.storages.vector.pgvector.filters import _convert_filters_to_query
+from dynamiq.types import Document
+from dynamiq.utils.logger import logger
+
+
+class PGVectorEnum(str, Enum):
+    def __contains__(cls, item: str) -> bool:
+        return item in [e.value for e in cls]
+
+    @classmethod
+    def list(cls):
+        return list(map(lambda c: c.value, cls))
+
+    @classmethod
+    def values(cls):
+        return list(cls)
+
+
+class PGVectorVectorFunction(PGVectorEnum):
+    COSINE_SIMILARITY = "cosine_similarity"
+    INNER_PRODUCT = "inner_product"
+    L2_DISTANCE = "l2_distance"
+    L1_DISTANCE = "l1_distance"
+
+
+class PGVectorIndexMethod(PGVectorEnum):
+    EXACT = "exact_nearest_neighbor_search"
+    IVFFLAT = "ivfflat"
+    HNSW = "hnsw"
+
+
+VECTOR_FUNCTION_TO_POSTGRESQL_OPS = {
+    PGVectorVectorFunction.COSINE_SIMILARITY: "vector_cosine_ops",
+    PGVectorVectorFunction.INNER_PRODUCT: "vector_ip_ops",
+    PGVectorVectorFunction.L2_DISTANCE: "vector_l2_ops",
+    PGVectorVectorFunction.L1_DISTANCE: "vector_l1_ops",
+}
+
+VECTOR_FUNCTION_TO_SCORE_DEFINITION = {
+    PGVectorVectorFunction.COSINE_SIMILARITY: "1 - (embedding <=> {query_embedding})",
+    PGVectorVectorFunction.INNER_PRODUCT: "(embedding <#> {query_embedding}) * -1",
+    PGVectorVectorFunction.L2_DISTANCE: "embedding <-> {query_embedding}",
+    PGVectorVectorFunction.L1_DISTANCE: "embedding <+> {query_embedding}",
+}
+
+
+class PGVectorStore:
+    """Vector store using pgvector."""
+
+    def __init__(
+        self,
+        connection: PGVector | str | None = None,
+        client: Optional["PGVector"] = None,
+        create_extension: bool = True,
+        table_name: str = "dynamiq_vector_store",
```

Comment:
I think `table` and `schema` names should be configurable in Nodes with default values (writer and retriever). So users could specify their own. Also make sense to be able to handle this in runtime similar to `content_key` and `embedding_key`.

Reply:
Should the usage for `table` and `schema` be the same as for `documents` in writer or `query_embedding`, `filters`, and `top_k` in retriever? Meaning if I should be able to extract the `table` and `schema` keys from the `input_data` dict and pass them to the vector store?

Reply:
table and schema should be configurable in the UI in the same way index is configured in Pinecone